diff -ru nano-2.2.6/src/nano.c nano-2.2.6+iPhone/src/nano.c
--- nano-2.2.6/src/nano.c	2010-11-12 06:13:08.000000000 +0000
+++ nano-2.2.6+iPhone/src/nano.c	2013-09-06 23:00:09.000000000 +0000
@@ -935,8 +935,7 @@
  * it was compiled with. */
 void version(void)
 {
-    printf(_(" GNU nano version %s (compiled %s, %s)\n"), VERSION,
-	__TIME__, __DATE__);
+    printf(_(" GNU nano version %s\n"), VERSION);
     printf(" (C) 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007,\n");
     printf(" 2008, 2009 Free Software Foundation, Inc.\n");
     printf(
