diff -ru vim71/src/os_unix.c vim71+iPhone/src/os_unix.c
--- vim71/src/os_unix.c	2007-05-09 19:41:58.000000000 +0000
+++ vim71+iPhone/src/os_unix.c	2008-01-01 12:27:46.000000000 +0000
@@ -753,12 +753,6 @@
     if (signal_stack != NULL)
     {
 # ifdef HAVE_SIGALTSTACK
-#  ifdef __APPLE__
-	/* missing prototype.  Adding it to osdef?.h.in doesn't work, because
-	 * "struct sigaltstack" needs to be declared. */
-	extern int sigaltstack __ARGS((const struct sigaltstack *ss, struct sigaltstack *oss));
-#  endif
-
 #  ifdef HAVE_SS_BASE
 	sigstk.ss_base = signal_stack;
 #  else
