diff -r -u tar-1.19/lib/argp-fmtstream.h tar-1.19+iPhone/lib/argp-fmtstream.h
--- tar-1.19/lib/argp-fmtstream.h	2007-09-28 13:11:36.000000000 +0000
+++ tar-1.19+iPhone/lib/argp-fmtstream.h	2007-12-26 23:04:38.000000000 +0000
@@ -135,15 +135,15 @@
 				      const char *__fmt, ...)
      __attribute__ ((__format__ (printf, 2, 3)));
 
-extern int __argp_fmtstream_putc (argp_fmtstream_t __fs, int __ch);
-extern int argp_fmtstream_putc (argp_fmtstream_t __fs, int __ch);
+static int __argp_fmtstream_putc (argp_fmtstream_t __fs, int __ch);
+static int argp_fmtstream_putc (argp_fmtstream_t __fs, int __ch);
 
-extern int __argp_fmtstream_puts (argp_fmtstream_t __fs, const char *__str);
-extern int argp_fmtstream_puts (argp_fmtstream_t __fs, const char *__str);
+static int __argp_fmtstream_puts (argp_fmtstream_t __fs, const char *__str);
+static int argp_fmtstream_puts (argp_fmtstream_t __fs, const char *__str);
 
-extern size_t __argp_fmtstream_write (argp_fmtstream_t __fs,
+static size_t __argp_fmtstream_write (argp_fmtstream_t __fs,
 				      const char *__str, size_t __len);
-extern size_t argp_fmtstream_write (argp_fmtstream_t __fs,
+static size_t argp_fmtstream_write (argp_fmtstream_t __fs,
 				    const char *__str, size_t __len);
 
 /* Access macros for various bits of state.  */
@@ -155,25 +155,25 @@
 #define __argp_fmtstream_wmargin argp_fmtstream_wmargin
 
 /* Set __FS's left margin to LMARGIN and return the old value.  */
-extern size_t argp_fmtstream_set_lmargin (argp_fmtstream_t __fs,
+static size_t argp_fmtstream_set_lmargin (argp_fmtstream_t __fs,
 					  size_t __lmargin);
 extern size_t __argp_fmtstream_set_lmargin (argp_fmtstream_t __fs,
 					    size_t __lmargin);
 
 /* Set __FS's right margin to __RMARGIN and return the old value.  */
-extern size_t argp_fmtstream_set_rmargin (argp_fmtstream_t __fs,
+static size_t argp_fmtstream_set_rmargin (argp_fmtstream_t __fs,
 					  size_t __rmargin);
 extern size_t __argp_fmtstream_set_rmargin (argp_fmtstream_t __fs,
 					    size_t __rmargin);
 
 /* Set __FS's wrap margin to __WMARGIN and return the old value.  */
-extern size_t argp_fmtstream_set_wmargin (argp_fmtstream_t __fs,
+static size_t argp_fmtstream_set_wmargin (argp_fmtstream_t __fs,
 					  size_t __wmargin);
 extern size_t __argp_fmtstream_set_wmargin (argp_fmtstream_t __fs,
 					    size_t __wmargin);
 
 /* Return the column number of the current output point in __FS.  */
-extern size_t argp_fmtstream_point (argp_fmtstream_t __fs);
+static size_t argp_fmtstream_point (argp_fmtstream_t __fs);
 extern size_t __argp_fmtstream_point (argp_fmtstream_t __fs);
 
 /* Internal routines.  */
@@ -198,7 +198,7 @@
 #endif
 
 #ifndef ARGP_FS_EI
-#define ARGP_FS_EI extern inline
+#define ARGP_FS_EI static inline
 #endif
 
 ARGP_FS_EI size_t
