diff -ru Python-2.5.1/setup.py Python-2.5.1+iPhone/setup.py
--- Python-2.5.1/setup.py	2007-02-14 12:53:41.000000000 +0000
+++ Python-2.5.1+iPhone/setup.py	2008-01-09 12:43:58.000000000 +0000
@@ -460,8 +430,7 @@
         # 64-bit platforms.
         exts.append( Extension('audioop', ['audioop.c']) )
 
-        # Disabled on 64-bit platforms
-        if sys.maxint != 9223372036854775807L:
+        if True:
             # Operations on images
             exts.append( Extension('imageop', ['imageop.c']) )
             # Read SGI RGB image files (but coded portably)
@@ -606,7 +577,7 @@
         # a release.  Most open source OSes come with one or more
         # versions of BerkeleyDB already installed.
 
-        max_db_ver = (4, 5)
+        max_db_ver = (4, 6)
         min_db_ver = (3, 3)
         db_setup_debug = False   # verbose debug prints from this script?
 
@@ -1025,8 +996,7 @@
                 exts.append(Extension('_codecs_' + loc,
                                       ['cjkcodecs/_codecs_%s.c' % loc]))
 
-        # Dynamic loading module
-        if sys.maxint == 0x7fffffff:
+        if True:
             # This requires sizeof(int) == sizeof(long) == sizeof(char*)
             dl_inc = find_file('dlfcn.h', [], inc_dirs)
             if (dl_inc is not None) and (platform not in ['atheos']):
