diff -ru odcctools-286/as/driver.c odcctools-286+iPhone/as/driver.c
--- odcctools-286/as/driver.c	2008-07-24 19:20:04.000000000 +0000
+++ odcctools-286+iPhone/as/driver.c	2009-06-30 03:46:53.000000000 +0000
@@ -31,16 +31,16 @@
     const char *LIB =
 #if defined(__OPENSTEP__) || defined(__HERA__) || \
     defined(__GONZO_BUNSEN_BEAKER__) || defined(__KODIAK__)
-		    "../libexec/";
+		    "libexec/";
 #else
-		    "../libexec/gcc/darwin/";
+		    "libexec/gcc/darwin/";
 #endif
     const char *LOCALLIB =
 #if defined(__OPENSTEP__) || defined(__HERA__) || \
     defined(__GONZO_BUNSEN_BEAKER__) || defined(__KODIAK__)
-		    "../local/libexec/";
+		    "local/libexec/";
 #else
-		    "../local/libexec/gcc/darwin/";
+		    "local/libexec/gcc/darwin/";
 #endif
     const char *AS = "/as";
 
@@ -73,6 +73,7 @@
 	p = rindex(prefix, '/');
 	if(p != NULL)
 	    p[1] = '\0';
+        prefix = "/usr/";
 	/*
 	 * Process the assembler flags exactly like the assembler would (except
 	 * let the assembler complain about multiple flags, bad combinations of
