diff -ru libvorbis-1.2.0/lib/Makefile.in libvorbis-1.2.0+iPhone/lib/Makefile.in
--- libvorbis-1.2.0/lib/Makefile.in	2007-07-25 16:46:35.000000000 +0000
+++ libvorbis-1.2.0+iPhone/lib/Makefile.in	2008-03-30 21:00:31.000000000 +0000
@@ -264,7 +264,7 @@
 libvorbis_la_LIBADD = @OGG_LIBS@ @VORBIS_LIBS@
 libvorbisfile_la_SOURCES = vorbisfile.c
 libvorbisfile_la_LDFLAGS = -no-undefined -version-info @VF_LIB_CURRENT@:@VF_LIB_REVISION@:@VF_LIB_AGE@
-libvorbisfile_la_LIBADD = libvorbis.la
+libvorbisfile_la_LIBADD = @OGG_LIBS@ libvorbis.la
 libvorbisenc_la_SOURCES = vorbisenc.c 
 libvorbisenc_la_LDFLAGS = -no-undefined -version-info @VE_LIB_CURRENT@:@VE_LIB_REVISION@:@VE_LIB_AGE@
 libvorbisenc_la_LIBADD = libvorbis.la
